core/src/main/java/com/linecorp/armeria/client/endpoint/dns/DnsAddressEndpointGroup.java

```diff
@@ -16,150 +16,188 @@
-     *     unspecified port, meaning the port will use a protocol-specific well-defined port number
-     *     (e.g., 80, 443).
-     * @param eventLoop the {@link EventLoop} to schedule DNS queries on.
-     * @param queryInterval the {@link Duration} to query DNS at.
-     */
-    public static DnsAddressEndpointGroup of(String hostname, int defaultPort, EventLoop eventLoop,
-                                             Duration queryInterval) {
-        return new DnsAddressEndpointGroup(hostname, defaultPort,
-                                           DnsEndpointGroupUtil.createResolverForEventLoop(eventLoop),
-                                           eventLoop, queryInterval);
-    }
-
-    private static final Logger logger = LoggerFactory.getLogger(DnsAddressEndpointGroup.class);
-
     private final String hostname;
-    private final int defaultPort;
-    private final NameResolver<InetAddress> resolver;
-    private final EventLoop eventLoop;
-    private final Duration queryInterval;
-
-    @Nullable
-    private ScheduledFuture<?> scheduledFuture;
-
-    @VisibleForTesting
-    DnsAddressEndpointGroup(String hostname, int defaultPort, NameResolver<InetAddress> resolver,
-                            EventLoop eventLoop, Duration queryInterval) {
-        checkArgument(defaultPort >= 0 && defaultPort <= 65535, "defaultPort must be between 0 and 65535");
-        this.hostname = requireNonNull(hostname, "hostname");
-        this.defaultPort = defaultPort;
-        this.resolver = requireNonNull(resolver, "resolver");
-        this.eventLoop = requireNonNull(eventLoop, "eventLoop");
-        this.queryInterval = requireNonNull(queryInterval, "queryInterval");
+    private final int port;
+
+    DnsAddressEndpointGroup(EventLoop eventLoop, int minTtl, int maxTtl,
+                            DnsServerAddressStreamProvider serverAddressStreamProvider,
+                            Backoff backoff, @Nullable ResolvedAddressTypes resolvedAddressTypes,
+                            String hostname, int port) {
+
+        super(eventLoop, minTtl, maxTtl, serverAddressStreamProvider, backoff,
+              newQuestions(hostname, resolvedAddressTypes),
+              resolverBuilder -> {
+                  if (resolvedAddressTypes != null) {
+                      resolverBuilder.resolvedAddressTypes(resolvedAddressTypes);
+                  }
+              });
+
+        this.hostname = hostname;
+        this.port = port;
+        start();
     }
 
-    /**
-     * Starts polling for service updates.
-     */
-    public void start() {
-        checkState(scheduledFuture == null, "already started");
-        scheduledFuture = eventLoop.scheduleAtFixedRate(this::query, 0, queryInterval.getSeconds(),
-                                                        TimeUnit.SECONDS);
+    private static List<DnsQuestion> newQuestions(String hostname, ResolvedAddressTypes resolvedAddressTypes) {
```

Comment:
Just curious. It seems that the returned list is empty if `resolvedAddressTypes` is `null`, and the constructor specifies `@Nullable ResolvedAddressTypes resolvedAddressTypes`. Is this intended?